tests/integration/container/tests/fastest_response_strategy.test.ts

```diff
@@ -0,0 +1,203 @@
+
+    driver = DriverHelper.getDriverForDatabaseEngine(env.engine);
+    initClientFunc = DriverHelper.getClient(driver);
+    await ProxyHelper.enableAllConnectivity();
+    client = null;
+    secondaryClient = null;
+    provider = null;
+    await TestEnvironment.verifyClusterStatus();
+    await TestEnvironment.verifyAllInstancesHasRightState("available");
+    await TestEnvironment.verifyAllInstancesUp();
+
+    RdsHostListProvider.clearAll();
+    PluginService.clearHostAvailabilityCache();
+  }, 1320000);
+
+  afterEach(async () => {
+    if (client !== null) {
+      try {
+        await client.end();
+      } catch (error) {
+        // pass
+      }
+    }
+    await PluginManager.releaseResources();
+    logger.info(`Test finished: ${expect.getState().currentTestName}`);
+  }, 1320000);
+
+  itIfMinThreeInstance(
+    "test failover to new reader use cached connection",
+    async () => {
+      // Connect to writer instance
+      const writerConfig = await initConfigWithFailover(
+        env.proxyDatabaseInfo.writerInstanceEndpoint,
+        env.proxyDatabaseInfo.instanceEndpointPort,
+        true
+      );
+      writerConfig["failoverMode"] = "reader-or-writer";
+      client = initClientFunc(writerConfig);
+
+      await client.connect();
+      const initialWriterId = await auroraTestUtility.queryInstanceId(client);
+      expect(await auroraTestUtility.isDbInstanceWriter(initialWriterId)).toStrictEqual(true);
+
+      await client.setReadOnly(true);
+
+      const readerConnectionId = await auroraTestUtility.queryInstanceId(client);
+      expect(readerConnectionId).not.toBe(initialWriterId);
+      // Get a reader instance
+      let otherReaderId;
+      for (const host of env.proxyDatabaseInfo.instances) {
+        if (host.instanceId && host.instanceId !== readerConnectionId && host.instanceId !== initialWriterId) {
+          otherReaderId = host.instanceId;
+          break;
+        }
+      }
+
+      if (!otherReaderId) {
+        throw new Error("Could not find a reader instance");
+      }
+      // Kill all instances except one other reader
+      for (const host of env.proxyDatabaseInfo.instances) {
+        if (host.instanceId && host.instanceId !== otherReaderId) {
+          await ProxyHelper.disableConnectivity(env.engine, host.instanceId);
+        }
+      }
+      await expect(async () => {
+        await auroraTestUtility.queryInstanceId(client);
+      }).rejects.toThrow(FailoverSuccessError);
+
+      const currentReaderId0 = await auroraTestUtility.queryInstanceId(client);
+
+      expect(currentReaderId0).toStrictEqual(otherReaderId);
+      expect(currentReaderId0).not.toBe(readerConnectionId);
+
+      await ProxyHelper.enableAllConnectivity();
+      await client.setReadOnly(false);
+
+      const currentId = await auroraTestUtility.queryInstanceId(client);
+      expect(currentId).toStrictEqual(initialWriterId);
+      // connect using cached fastest connection
```

Comment:
```suggestion
      // Connect using cached fastest connection.
```

nit: format comments with a capital and a period